### DISCUSSION TITLE
considering adding correct formating a requirement

### DISCUSSION THREAD

**Discussion by oroulet** [FreeOpcUa/opcua-asyncio#413] [Ideas]
maybe use dark or yapf as reference and use flake8. 
My only issue is that I hate short lines, so if we do that I will probably remove line length limitation or set it to 160 for example

And then make it clear that yapf/black is adviced before submitting a PR

A related task would be to use black on all existing code. that will be a quite intrusive commit... Not sure it should be done..

https://www.kevinpeters.net/auto-formatters-for-python

**Comment by ZuZuD** [comment]
FYI PEP8 recommends [79 characters per-line](https://www.python.org/dev/peps/pep-0008/#maximum-line-length). Suggesting the contributors to use an auto-formatter when the repository doesn't fully comply might be complicated since they usually mutate a whole file.

If the repository was compliant though, the linter could be enforced during PR pre-flight checks and that would be nice.

**Reply by AndreasHeine** [reply]
https://pypi.org/project/pep8/ ???

**Reply by swamper123** [reply]
It is possible to add them into the github actions as well and if it fails, it fails. You may [take a look on that yaml file ](https://github.com/gijzelaerr/python-snap7/blob/master/.github/workflows/flake8.yml)in a project. It is short, understandable and may need just a few changes. (I would like to add [MyPy ](http://mypy-lang.org/)or [Bandit ](https://github.com/PyCQA/bandit)as well, but I know everyone would hate it. ^^ Perhaps in a later version.)

On the other hand a general "make over" in styling may be neccessarry for a better base...but there will be many.
Result from pycharms inspect code(default, with PythonSecurity incl.), included folders: asyncua(except namespaces parts), docs, tests, tools, ua:
![grafik](https://user-images.githubusercontent.com/24408657/105459734-0c35e400-5c8b-11eb-8f03-f1fa2441d7f8.png)
![grafik](https://user-images.githubusercontent.com/24408657/105459763-1526b580-5c8b-11eb-9371-bc172af86be7.png)

So round about 2900 python style issues and in sum around 3400 issues, which have to be solved first, before we may add it in actions (and just take the important files, not the whole namespaces incl.).

We can discuss about line length if it should be 78/79/120 or longer...but I see that as a minor point at the moment.

**Reply by oroulet** [reply]
if we exclude autogenerated files the list of errors is quite small.
```
 flake8 asyncua | grep -v auto | grep -v standard | wc -l
75
```
and most of them are small formating things that black and yapf will fix automatically

rmq: 75 errors only but my config of flake8 disable line length checking since we disabled it at work

**Reply by oroulet** [reply]
@ZuZuD PEP8 also says that 99 is fine. and that 80 char line length limitation is often discussed as archaic.
At work we have no limitation but ask people to make something that looks good and it works quite fine, but I check now and it looks like people tend to stay under 100.
My editor is also 100 lines width I see now...

**Comment by oroulet** [comment]
@swamper123 we already have the flake8 stuff in prepository. We jsut need to remove the --exit.-zero and maybe add some config https://github.com/FreeOpcUa/opcua-asyncio/blob/master/.github/workflows/python-package.yml#L36

**Comment by oroulet** [comment]
OK. I did some tests and changed my mind. I think the cleaner is use Black without any configuraiton as it is meant to be used. that means 88 char line length. I will probably do it one day and reformat everything unless someone comes with a good argument against.
Black has good info about how to handle `git blame` and others: https://pypi.org/project/black/
